packages/unlock-js/src/web3Service.ts

```diff
@@ -428,22 +428,39 @@ export default class Web3Service extends UnlockService {
    * @param {Number} network
    */
   async totalKeys(lockAddress: string, owner: string, network: number) {
-    const version = await this.lockContractAbiVersion(
+    const contract = await this.getLockContract(
       lockAddress,
       this.providerForNetwork(network)
     )
-
-    if (!version.totalKeys) {
-      throw new Error('Lock version not supported')
+    const publicLockVersion = await contract.publicLockVersion()
+    if (!contract.totalKeys) {
```

Comment:
Let's keep that logic inside of the version files as it is much easier to cleanup/maintain please.